src/dimred/lsi/test.py

```diff
@@ -0,0 +1,206 @@
+import sys
+import pytest
+import subprocess
+import mudata as mu
+import numpy as np
+
+## VIASH START
+meta = {
+    'resources_dir': 'resources_test',
+    'executable': './target/docker/dimred/lsi/lsi',
+    'config': './src/dimred/lsi/config.vsh.yaml'
+}
+## VIASH END
+
+input_path = f"{meta['resources_dir']}/LSI/test_data.h5mu"
+
+
+'''
+Tests: 
+1. general test 
+2. test HVF
+3. test modality
+4. test layer
+5. test overwrite 
+'''
+
+
+
+# 1.general test
+def test_lsi(tmp_path):
+    output_path = tmp_path / "output_lsi.h5mu"
+
+    # run component
+    cmd_args = [
+    meta["executable"],
+     "--input", input_path,
+     "--output", str(output_path),
+     "--obsm_output", "X_test",
+     "--num_components", "30"
+    ]
+    subprocess.run(cmd_args, check=True)
+    
+
+    assert output_path.is_file()
+    data = mu.read_h5mu(output_path)
+    assert "X_test" in data.mod['atac'].obsm
+    assert data.mod["atac"].obsm["X_test"].shape == (data.mod["atac"].n_obs, 30)
+    assert "lsi" in data.mod['atac'].uns
+    assert "LSI" in data.mod['atac'].varm
+
+
+
+# 2.test HVF 
+def test_select_highly_variable_column(tmp_path):
+    output_path = tmp_path / "output_lsi.h5mu"
+
+    # run component
+    cmd_args = [
+    meta["executable"],
+     "--input", str(input_path),
+     "--output", str(output_path),
+     "--var_input", "highly_variable"
+    ]
+    subprocess.run(cmd_args, check=True)
+    
+
+    assert output_path.is_file()
+    data = mu.read_h5mu(output_path)
+    assert "X_lsi" in data.mod['atac'].obsm
+    assert data.mod["atac"].obsm["X_lsi"].shape == (data.mod["atac"].n_obs, 50)
+    assert "highly_variable" in data.mod["atac"].var.columns
+    assert "lsi" in data.mod['atac'].uns
+    assert "LSI" in data.mod['atac'].varm
+    assert data.mod["atac"].varm["LSI"].shape == (data.mod["atac"].n_vars, 50)
+
+
+def test_highly_variable_column_does_not_exist_raises():
+    with pytest.raises(subprocess.CalledProcessError) as err:
```

Comment:
Interesting approach to testing :) If it works, that's fine, otherwise `with pytest.raises(ValueError)` should be enough